Code/Tools/ProjectManager/Source/PythonBindingsInterface.h

```diff
@@ -88,6 +88,15 @@ namespace O3DE::ProjectManager
         virtual DetailedOutcome SetEngineInfo(const EngineInfo& engineInfo, bool force = false) = 0;
 
 
+        // Remote source
+
+        /**
+         * Validates a repository without adding it.
+         * @param repoUri the absolute filesystem path or url to the repo.
+         * @return bool, true if the repository is valid
+         */
+        virtual bool ValidateRepository(const QString& repoUri) = 0;
```

Comment:
This function should be marked `const`

Reply:
+1

Reply:
This function calls ExecuteWithLock which is not const due to clearing/possibly writing to local copies of error strings returned from Python.

Reply:
Shouldn't ExecuteWithLock be marked `const` with the PythonBindings::m_lock mutex marked `mutable`?
Semantically ExecuteWithLock is const function as it doesn't modify the state of the python bindings.